(develop) Compiler warns regarding initialization order of class fields

```
../jubatus/server/framework/mixer/linear_mixer.hpp: In constructor ‘jubatus::server::framework::mixer::linear_mixer::linear_mixer(jubatus::util::lang::shared_ptr<jubatus::server::framework::mixer::linear_communication>, jubatus::util::concurrent::rw_mutex&, unsigned int, unsigned int)’:
../jubatus/server/framework/mixer/linear_mixer.hpp:122:40: warning: ‘jubatus::server::framework::mixer::linear_mixer::model_mutex_’ will be initialized after [-Wreorder]
../jubatus/server/framework/mixer/linear_mixer.hpp:109:16: warning:   ‘unsigned int jubatus::server::framework::mixer::linear_mixer::count_threshold_’ [-Wreorder]
../jubatus/server/framework/mixer/linear_mixer.cpp:242:1: warning:   when initialized here [-Wreorder]
```

```
../jubatus/server/framework/mixer/push_mixer.hpp: In constructor ‘jubatus::server::framework::mixer::push_mixer::push_mixer(jubatus::util::lang::shared_ptr<jubatus::server::framework::mixer::push_communication>, jubatus::util::concurrent::rw_mutex&, unsigned int, unsigned int, const std::pair<std::basic_string<char>, int>&)’:
../jubatus/server/framework/mixer/push_mixer.hpp:125:40: warning: ‘jubatus::server::framework::mixer::push_mixer::model_mutex_’ will be initialized after [-Wreorder]
../jubatus/server/framework/mixer/push_mixer.hpp:113:16: warning:   ‘unsigned int jubatus::server::framework::mixer::push_mixer::count_threshold_’ [-Wreorder]
../jubatus/server/framework/mixer/push_mixer.cpp:188:1: warning:   when initialized here [-Wreorder]
```

--- a/jubatus/server/framework/mixer/linear_mixer.cpp
+++ b/jubatus/server/framework/mixer/linear_mixer.cpp
@@ -245,14 +245,14 @@ linear_mixer::linear_mixer(
     unsigned int count_threshold,
     unsigned int tick_threshold)
     : communication_(communication),
-      model_mutex_(mutex),
       count_threshold_(count_threshold),
       tick_threshold_(tick_threshold),
       counter_(0),
       ticktime_(get_clock_time()),
       is_running_(false),
       is_obsolete_(true),
-      t_(jubatus::util::lang::bind(&linear_mixer::stabilizer_loop, this)) {
+      t_(jubatus::util::lang::bind(&linear_mixer::stabilizer_loop, this)),
+      model_mutex_(mutex) {
 }
 
 linear_mixer::~linear_mixer() {

--- a/jubatus/server/framework/mixer/push_mixer.cpp
+++ b/jubatus/server/framework/mixer/push_mixer.cpp
@@ -192,7 +192,6 @@ push_mixer::push_mixer(
     unsigned int tick_threshold,
     const std::pair<std::string, int>& my_id)
     : communication_(communication),
-      model_mutex_(mutex),
       count_threshold_(count_threshold),
       tick_threshold_(tick_threshold),
       my_id_(my_id),
@@ -200,7 +199,8 @@ push_mixer::push_mixer(
       mix_count_(0),
       ticktime_(get_clock_time()),
       is_running_(false),
-      t_(jubatus::util::lang::bind(&push_mixer::mixer_loop, this)) {
+      t_(jubatus::util::lang::bind(&push_mixer::mixer_loop, this)),
+      model_mutex_(mutex) {
 }
 
 push_mixer::~push_mixer() {

